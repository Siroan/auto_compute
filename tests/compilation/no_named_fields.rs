extern crate compute;

#[macro_use]
extern crate compute_macro;

fn main() {
    #[derive(Equation)]
    struct TupleStruct(i32, i32);
}