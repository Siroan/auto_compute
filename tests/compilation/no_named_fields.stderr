error: The equation must contain named fields
 --> tests/compilation/no_named_fields.rs:7:14
  |
7 |     #[derive(Equation)]
  |              ^^^^^^^^
  |
  = note: this error originates in the derive macro `Equation` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `auto_compute` found for reference `&TupleStruct` in the current scope
 --> tests/compilation/no_named_fields.rs:7:14
  |
7 |     #[derive(Equation)]
  |              ^^^^^^^^
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following trait defines an item `auto_compute`, perhaps you need to implement it:
          candidate #1: `EquationAutoCompute`
  = note: this error originates in the derive macro `Equation` (in Nightly builds, run with -Z macro-backtrace for more info)
